### DISCUSSION TITLE
Screen mirroring function

### DISCUSSION THREAD

**Discussion by Madpat88** [vchlum/nano-lights#4] [Ideas]
Do you think it will be possible to implement the screen mirroring function as is available in the Mac/Windows desktop app?

**Comment by vchlum** [comment]
Hi @Madpat88 ! Yes, it is possible. The convenient way how to do it is by GStreamer. Unfortunately, there is the bug: https://gitlab.gnome.org/GNOME/gjs/-/issues/557 Once this bug is resolved, I can proceed with implementation.

**Comment by vchlum** [comment]
Hi @Madpat88 !

I created a new extension 'Smart Home'. You can find it here:
 * https://extensions.gnome.org/extension/7737/smart-home/
 * https://github.com/vchlum/smart-home
This new extension supports not only Nanoleaf but also Philips Hue, Ikea Dirigera, Shelly, and Home Assistant. Nano Lights extension is deprecated now. I will continue to develop only the new extension.

The new extension has supported screen sync/mirroring with Nanoleaf devices since version 9. The feature mirrors the screen (or selected display) to the lights. You need to set the correct panel orientation in the original app. You can find the item "Screen" or "Display X" (in case of more than one display) between the scenes.

Check it out and let me know if it works as expected.
